Utils/ScriptValidator/Unit1.pas

```diff
@@ -53,45 +73,128 @@ procedure TForm1.FormCreate(Sender: TObject);
 procedure TForm1.FormDestroy(Sender: TObject);
 begin
   fScripting.Free;
+  sListFileInFolder.Free;
   DragAcceptFiles(Handle, False);
 end;
 
-procedure TForm1.btnBrowseClick(Sender: TObject);
+procedure TForm1.btnBrowseFileClick(Sender: TObject);
 begin
   if OpenDialog1.Execute then
-    Edit1.Text := OpenDialog1.FileName;
+  Edit1.Text := OpenDialog1.FileName;
 end;
 
 
+procedure TForm1.btnBrowsePathClick(Sender: TObject);
+var DirValidate : String;
+begin
+  if not SelectDirectory('Select the folder to Validate scripts', '', DirValidate) then Exit;
+  Edit1.Text := DirValidate;
+end;
+
+procedure TForm1.Edit1Change(Sender: TObject);
+begin
+  If (GetFileAttributes(PWideCHar(TEdit(Sender).Text)) and FILE_ATTRIBUTE_DIRECTORY) = FILE_ATTRIBUTE_DIRECTORY then
+  begin
+    IsValidatePath := true;
+    Memo1.Text := 'Selected folders!';
+  end
+  else
+  begin
+    IsValidatePath := false;
+    Memo1.Text := 'Select Files!';
+  end;
+end;
+
 procedure TForm1.btnValidateClick(Sender: TObject);
+var I : Integer;
 begin
-  Validate(Edit1.Text, True);
+  btnBrowsePath.Enabled := false;
+  btnBrowseFile.Enabled := false;
+  btnValidate.Enabled := false;
+  btnValidateAll.Enabled := false;
+  Edit1.Enabled := false;
+
+  Memo1.Lines.Clear;
+  if IsValidatePath then
+  begin
+    if SameText(RightStr(Edit1.Text, 1), PathDelim) then Edit1.Text := LeftStr(Edit1.Text, (Length(Edit1.Text) -1));
+
+    if not DirectoryExists(Edit1.Text) then
+      Memo1.Lines.Append('Directory not found ' + Edit1.Text)
+    else
+    begin
+      Memo1.Lines.Append('Search for files in a folder ...');
+      sListFileInFolder.Clear;
+      FindFiles(Edit1.Text, sListFileInFolder);
+      Memo1.Lines.Append('Check '+Edit1.Text);
+      if sListFileInFolder.Count = 0 then
+        Memo1.Lines.Append('No files in a directory ' + Edit1.Text)
+      else
+      begin
+        Memo1.Lines.Append('Files in the folder: '+IntToStr(sListFileInFolder.Count));
+        for I := 0 to sListFileInFolder.Count-1 do
+          Validate(sListFileInFolder.Strings[I], True);
+        Memo1.Lines.Append('Checked ' + IntToStr(sListFileInFolder.Count) + ' in ..\..\');
+      end;
+
+    end;
+
+  end else
+    Validate(Edit1.Text, True);
+
+  btnBrowsePath.Enabled := true;
+  btnBrowseFile.Enabled := true;
+  btnValidate.Enabled := true;
+  btnValidateAll.Enabled := true;
+  Edit1.Enabled := true;
 end;
 
 
-procedure TForm1.Button1Click(Sender: TObject);
+procedure TForm1.btnValidateAllClick(Sender: TObject);

```

Comment:
Below function looks like to have a lot of duplicate code with above. Are you sure they can not be merged?